core/src/backend/navigator.rs

```diff
@@ -389,3 +389,14 @@ impl NavigatorBackend for NullNavigatorBackend {
         url
     }
 }
+
+pub fn is_fetch_domain_banned(url: &str) -> bool {
+    if let Ok(parsed_url) = Url::parse(url) {
+        if let Some(domain) = parsed_url.domain() {
+            if domain.ends_with("mochiads.com") {
```

Comment:
This would also block the domain `examplemochiads.com`; should probably be `domain == "mochiads.com" || domain.ends_with(".mochiads.com")` or similar

Reply:
Some other things to keep in mind (although these are most likely nitpicking, overly defensive, and not relevant for this scenario):
- `mochiads.com.` (note the trailing period) is the same as `mochiads.com`, except it's "absolute" instead of "relative" (thus, arguably more correct anyway, except almost nobody uses this form anywhere)
- International domain names may be Punycode encoded, or represented directly in Unicode - not sure where the en-/decoding would occur, if any.

Reply:
I guess this is just the job for something like `tldextract` crate.

Reply:
Punycoding an ASCII domain merely prepends `xn--` and appends `-` to the domain. The idea is that the prefix says "this is a Punycode domain", then you put all of the directly representable characters in the string, then the `-`, and finally an encoded list of Unicode code points and where to insert them into the ASCII part of the domain.

Strictly speaking, `mochiads.com` and `xn--mochiads-.com` are separate domains. I don't know if/how Adobe handled Punycoding internally or if movie authors were expected to do it themselves. I'm going to assume it works like HTML5 fetch, and thus `is_fetch_domain_banned` should be seeing Unicode strings that get passed to HTML5 fetch and then the browser actually encodes any IDNs for us. Please correct me if I'm wrong.